### ISSUE TITLE
Memory Issue when uploading large files

### ISSUE THREAD

**Issue by vitalielias** [kit-data-manager/mapping-service#15]
We need the capability of processing several dozens (or even hundreds) of TIFFs at a time. Currently, the capability is unknown, but it is very likely limited by the memory allocation of the mapping service on BW Cloud.

I tried uploading a zip file containing 700+ TIFF images. It is 823 MB compressed, and 1.27 GB uncompressed. Currently, the live mapping service has problems handling the zip file, it outputs the following results file:

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>502 Bad Gateway</title>
</head><body>
<h1>Bad Gateway</h1>
<p>The proxy server received an invalid
response from an upstream server.<br />
</p>
<hr>
<address>Apache/2.4.53 (Debian) Server at metarepo.nffa.eu Port 443</address>
</body></html>
```

which suggests it simply timed out trying to process all of the images.

**Comment by VolkerHartmann** [comment]
Is posting hundreds of MB of data to a service a good idea? 
SpringBoot can limit the size of data to make DOS attacks more difficult. 
Allowing almost arbitrarily large files can easily break the server. 
Since the client has to zip all the files anyway and then decompress them at the end, it's probably easier for the client to send them one at a time.
If you want to process multiple files at once you should use asynchronous REST requests.

**Comment by vitalielias** [comment]
As this wasn't an issue on my local machine, which has more than enough memory availability, we figured there must be some issue with the BW cloud instance of the mapping service.

As a fix, a virtual machine with much more memory was set up by @VolkerHartmann and an instance of the mapping service was installed. This can be [accessed here](http://mapping.datamanager.kit.edu:8095/) (must be on KIT network). The setup steps are documented [here](https://git.scc.kit.edu/dem/whiteboard/-/wikis/Server@DEM/Mapping).

The `/build/resources/main/application.propertiesapplication.properties` config file of the mapping service also needed to be edited. Particularly:

```
spring.servlet.multipart.max-file-size=1024MB
spring.servlet.multipart.max-request-size=1024MB
```

were changed from their default values, which was 100MB.

Surface level tests showed uploading large zip files did not cause the service to crash after these fixes were implemented, but the actual results could not be determined as a plugin for the large dataset which was used to set is not yet available. This will also be tested in the coming days, but should be a topic of conversation for the matwerk data manager instance of the mapping service as well.
